### PR TITLE
Add params to define fiopush/check value

### PR DESCRIPTION
**Pull request by mike-sul**
New params to specify fiopush/fiocheck to use, by default or if the specified one are not valid/exist then fiopush/fiocheck built by OE are used.

Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

**Comment by mike-sul** [comment]
https://github.com/foundriesio/ci-scripts/pull/175/files and https://github.com/foundriesio/lmp-manifest/pull/178 will make LmP builds use the container's fiotools (lmp-sdk).
I verified all use-cases.

**Review by doanac — COMMENTED** [review]
i think there's some tabs vs spaces formatting that may be happening.

Maybe `run_fiotool_cmd` shouldn't take any arguments. Then you can have that bit of
```
# Fallback to the OE built fiocheck if FIO_CHECK_CMD is not defined or it refers to non-existing or non-executable
```
in a single place

**Comment by mike-sul** [comment]
> i think there's some tabs vs spaces formatting that may be happening.

It looks like the whole file have had mixed style, I'll fix it.

**Comment by mike-sul** [comment]
> Maybe `run_fiotool_cmd` shouldn't take any arguments. Then you can have that bit of

I am not sure I get it, there are two use-cases, push and check, without any arguments how the function will know which of the commands to run and what's the fallback command?

File: meta-lmp-base/classes/lmp.bbclass [hunk 1]
```diff
@@ -47,53 +51,61 @@ IMAGE_CMD:ostree:append () {
...
-        if not (dtb.endswith('dtb') or dtb.endswith('dtbo')):
-            bb.error("KERNEL_DEVICETREE entry %s is not a .dtb or .dtbo file." % (dtb) )
-        return os.path.basename(dtb) + ';dtb/' + os.path.basename(dtb)
+	def transform(dtb):
+		if not (dtb.endswith('dtb') or dtb.endswith('dtbo')):
+			bb.error("KERNEL_DEVICETREE entry %s is not a .dtb or .dtbo file." % (dtb) )
+		return os.path.basename(dtb) + ';dtb/' + os.path.basename(dtb)
 
-    return ' '.join([transform(dtb) for dtb in alldtbs.split() if dtb])
+	return ' '.join([transform(dtb) for dtb in alldtbs.split() if dtb])
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp.bbclass:111]: Can you drop these changes (in make_efi_dtb_boot_files) as part of this patch? Fine to have it done isolated in another patch, but we also need to make sure this won't cause problems as the logic is done in python.

File: meta-lmp-base/classes/lmp.bbclass [hunk 2]
```diff
@@ -47,53 +51,61 @@ IMAGE_CMD:ostree:append () {
...
-                fi
-        else
-                bbwarn "SOTA_PACKED_CREDENTIALS not set. Please add SOTA_PACKED_CREDENTIALS."
-        fi
+		else
+			bbwarn "SOTA_PACKED_CREDENTIALS file does not exist."
+		fi
+	else
+		bbwarn "SOTA_PACKED_CREDENTIALS not set. Please add SOTA_PACKED_CREDENTIALS."
+	fi
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp.bbclass:71]: Please also fix this indentation issues in a separated patch/commit.

File: meta-lmp-base/classes/lmp.bbclass (hunk 1 above)
Comment (mike-sul) [reply meta-lmp-base/classes/lmp.bbclass]: I've dropped them. Can make this fix in another commit or PR.

File: meta-lmp-base/classes/lmp.bbclass (hunk 2 above)
Comment (mike-sul) [reply meta-lmp-base/classes/lmp.bbclass:72]: Done.

File: meta-lmp-base/classes/lmp.bbclass [hunk 3]
```diff
@@ -64,17 +75,17 @@ run_fiotool_cmd () {
 do_image_ostreepush[depends] += "ostreeuploader-native:do_populate_sysroot"
 IMAGE_CMD:ostreepush:prepend () {
 	if [ "${USE_FIOTOOLS}" = "1" ]; then
-		run_fiotool_cmd "fiopush"
-		# force return so garage-push called from meta-updater's IMAGE_CMD:ostreepush is not executed
+		run_fiotool_cmd "${FIO_PUSH_CMD}" "fiopush"
+		# force return so garage-push called from meta-updater's IMAGE_CMD_ostreepush is not executed
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp.bbclass:79]: No need to change this comment line, the correct syntax now (>= honister) is indeed `IMAGE_CMD:ostreepush`.

File: meta-lmp-base/classes/lmp.bbclass [hunk 4]
```diff
@@ -64,17 +75,17 @@ run_fiotool_cmd () {
...
 	fi
 }
 
 do_image_garagecheck[depends] += "ostreeuploader-native:do_populate_sysroot"
 IMAGE_CMD:garagecheck:prepend () {
 	if [ "${USE_FIOTOOLS}" = "1" ]; then
-		run_fiotool_cmd "fiocheck"
-		# force return so garage-check called from meta-updater's IMAGE_CMD:garagecheck is not executed
+		run_fiotool_cmd "${FIO_CHECK_CMD}" "fiocheck"
+		# force return so garage-check called from meta-updater's IMAGE_CMD_garagecheck is not executed
```
Comment (ricardosalveti) [inline meta-lmp-base/classes/lmp.bbclass:88]: Same here.

File: meta-lmp-base/classes/lmp.bbclass (hunk 3 above)
Comment (mike-sul) [reply meta-lmp-base/classes/lmp.bbclass:79]: wtf, I don't recall changing it :).

Comment (mike-sul) [reply meta-lmp-base/classes/lmp.bbclass]: done

Comment (ricardosalveti) [reply meta-lmp-base/classes/lmp.bbclass]: Probably because you changed this in hardknott and moved back the file to honister? But all good now :-)

**Review by ricardosalveti — APPROVED** [review]
LGTM
